### PR TITLE
Enhanced reporting

### PR DESCRIPTION
**Pull request by eddynaka**

### CODE REVIEW

File: Src/Sarif.PatternMatcher/AnalyzeOptions.cs [hunk 1]
```diff
@@ -43,7 +43,14 @@ public class AnalyzeOptions : AnalyzeOptionsBase
...
         [Option(
-            "disable-dynamic-validation-caching")]
+            "disable-dynamic-validation-caching",
+            HelpText = "Disable caching from dynamic validation.")]
         public bool DisableDynamicValidationCaching { get; internal set; }
+
+        [Option(
+            "enhanced-reporting",
+            HelpText = "Enable enhanced reporting. If possible, it will fetch data from " +
+                       "the provider and add more information to the reporting message.")]
```
Comment (michaelcfanning) [inline Src/Sarif.PatternMatcher/AnalyzeOptions.cs:53]: "Enable enhanced reporting provided by dynamic validators (when --dynamic-validation is also enabled)."

File: Src/Sarif.PatternMatcher/ValidatingVisitor.cs [hunk 2]
```diff
@@ -37,12 +39,14 @@ public override Result VisitResult(Result node)
                 // Our validation messages currently look like so.
                 // {0:scanTarget}' contains {1:validationPrefix}{2:encoding}{3:secretKind}{4:validationSuffix}{5:validatorMessage}
                 string message = null;
+                IDictionary<string, string> options = new Dictionary<string, string>();
```
Comment (michaelcfanning) [inline Src/Sarif.PatternMatcher/ValidatingVisitor.cs:42]: >IDictionary<string, string> [](start = 16, length = 27)

use 'var'

**Review by michaelcfanning — APPROVED** [review]
:shipit:

File: Src/Sarif.PatternMatcher/ValidatingVisitor.cs (hunk 2 above)
Comment (eddynaka) [reply Src/Sarif.PatternMatcher/ValidatingVisitor.cs:42]: the method below needs IDictionary, so we need to enforce using the interface

**Pull request merged**